dataService.js

```diff
@@ -0,0 +1,71 @@
+// Required Maps
+let productMap;
+let categoryMap;
+let combinedProductMap;
+
+/*
+ * Method to create and return map of products if its not already created
+ */
+function getProducts() {
+
+  if (!productMap || !productMap.size) {
```

Comment:
What do you think about this change on this line?
```suggestion
  if (!(productMap && productMap.size)) {
```

Reply:
Yes, that will be more explicit. Done!